### PR TITLE
c++ default IMU axis aligned to camera RF

### PR DESCRIPTION
**Pull request by arrenglover**
index the IMU correctly such that it is aligned to the camera image plane

### CODE REVIEW

File: lib/src/codecs/codec_IMUEvent.cpp [hunk 1]
```diff
@@ -86,4 +86,38 @@ std::string IMUevent::getType() const
...
+
+
+    switch(index) {
+    case(ACC_X):
+        return (value * (9.80665 / 16384.0)+ax_bias)*ax_gain;
+    case(ACC_Y):
+        return (value * -(9.80665 / 16384.0)+ay_bias)*ay_gain;
+    case(ACC_Z):
+        return (value * (9.80665 / 16384.0)+az_bias)*az_gain;
+    case(GYR_X):
```
Comment (simbamford) [inline lib/src/codecs/codec_IMUEvent.cpp:106]: Missing a return value here

File: lib/src/codecs/codec_IMUEvent.cpp [hunk 2]
```diff
@@ -86,4 +86,38 @@ std::string IMUevent::getType() const
...
+    case(GYR_Z):
+        return value * (250.0 * M_PI / (2.0 * 180.0 * 16384.0));
+    case(GYR_Y):
+        return value * -(250.0 * M_PI / (2.0 * 180.0 * 16384.0));
+    case(TEMP):
+        return (value + (273.15 - 21)) * 333.87;
+    case(MAG_X):
```
Comment (simbamford) [inline lib/src/codecs/codec_IMUEvent.cpp:113]: Missing a return value also here

File: lib/src/codecs/codec_IMUEvent.cpp [hunk 3]
```diff
@@ -86,4 +86,38 @@ std::string IMUevent::getType() const
...
+        return (value * -(9.80665 / 16384.0)+ay_bias)*ay_gain;
+    case(ACC_Z):
+        return (value * (9.80665 / 16384.0)+az_bias)*az_gain;
+    case(GYR_X):
+    case(GYR_Z):
+        return value * (250.0 * M_PI / (2.0 * 180.0 * 16384.0));
+    case(GYR_Y):
+        return value * -(250.0 * M_PI / (2.0 * 180.0 * 16384.0));
+    case(TEMP):
+        return (value + (273.15 - 21)) * 333.87;
```
Comment (simbamford) [inline lib/src/codecs/codec_IMUEvent.cpp:112]: Should be +21 deg I think.

File: lib/src/vDraw_basic.cpp [hunk 4]
```diff
@@ -241,40 +241,40 @@ void imuDraw::draw(cv::Mat &image, const ev::vQueue &eSet, int vTime)
...
             break; }
-        case 1: {//acc y
-            auto p_end = centre + cv::Point(aep->value * lin_scaler, 0);
+        case IMUevent::ACC_X: {//acc x
+            auto p_end = centre - cv::Point(aep->value * lin_scaler, 0);
             cv::line(image, centre, p_end, violet, 4);
             break;}
-        case 2: {//acc z
+        case IMUevent::ACC_Z: {//acc z
             auto p_end = centre + cv::Point(aep->value * lin_scaler * 0.71,
```
Comment (simbamford) [inline lib/src/vDraw_basic.cpp:253]: Is this a z value being split between x and y in a 2d point structure? Why?

File: lib/src/codecs/codec_IMUEvent.cpp [hunk 5]
```diff
@@ -86,4 +86,38 @@ std::string IMUevent::getType() const
     return IMUevent::tag;
 }
 
+double IMUevent::convertToSI(int index, int value)
+{
+    static constexpr double ay_bias = 0.12;
```
Comment (simbamford) [inline lib/src/codecs/codec_IMUEvent.cpp:91]: Bias and gains coming from individual sensor calibration - would be better to bring those in via a config file.

Comment (arrenglover) [reply lib/src/codecs/codec_IMUEvent.cpp:91]: agreed. that won't happen in this PR though and should be done in a second update when required

File: lib/src/codecs/codec_IMUEvent.cpp (hunk 3 above)
Comment (arrenglover) [reply lib/src/codecs/codec_IMUEvent.cpp:112]: I think I changed this on purpose from your values as I found -21 was correct

File: lib/src/codecs/codec_IMUEvent.cpp (hunk 1 above)
Comment (arrenglover) [reply lib/src/codecs/codec_IMUEvent.cpp:106]: GYR_X falls through to GYR_Z as they have the same function

File: lib/src/codecs/codec_IMUEvent.cpp (hunk 2 above)
Comment (arrenglover) [reply lib/src/codecs/codec_IMUEvent.cpp:113]: MAG_X falls through to MAG_Z as they have the same function

File: lib/src/vDraw_basic.cpp (hunk 4 above)
Comment (arrenglover) [reply lib/src/vDraw_basic.cpp:253]: just for visualisation - that makes the 3rd axis "into" the screen.

**Pull request merged**